### PR TITLE
added spec version to metadata

### PR DESCRIPTION
**Pull request by tulshi**

### CODE REVIEW

File: openid-sharedsignals-framework-1_0.md [hunk 1]
```diff
@@ -563,6 +568,28 @@ configuration information.
 
 Transmitters have metadata describing their configuration:
 
+spec_version
+
+> OPTIONAL. A version idenitfying the implementer's draft or final specification implemented by the Transmitter. This includes the numerical portion of the spec version as described in the document {{NAMINGCONVENTION}}. If absent, the Transmitter is assumed to conform to "1_0-ID1" version of the specification.
```
Comment (FragLegs) [inline openid-sharedsignals-framework-1_0.md:573]:
```suggestion
> OPTIONAL. A version identifying the implementer's draft or final specification implemented by the Transmitter. This includes the numerical portion of the spec version as described in the document {{NAMINGCONVENTION}}. If absent, the Transmitter is assumed to conform to "1_0-ID1" version of the specification.
```

**Review by FragLegs — APPROVED** [review]

File: openid-sharedsignals-framework-1_0.md [hunk 2]
```diff
@@ -563,6 +568,28 @@ configuration information.
 
 Transmitters have metadata describing their configuration:
 
+spec_version
+
+> OPTIONAL. A version identifying the implementer's draft or final specification implemented by the Transmitter. This includes the numerical portion of the spec version as described in the document {{NAMINGCONVENTION}}. If absent, the Transmitter is assumed to conform to "1_0-ID1" version of the specification.
```
Comment (timcappalli) [inline openid-sharedsignals-framework-1_0.md:573]: maybe for clarity, add to the end "(this document)"

so: "If absent, the Transmitter is assumed to conform to "1_0-ID1" version of the specification (this document)."

**Review by timcappalli — CHANGES_REQUESTED** [review]
minor change

File: openid-sharedsignals-framework-1_0.md (hunk 2 above)
Comment (tulshi) [reply openid-sharedsignals-framework-1_0.md]: Updated.

**Review by timcappalli — APPROVED** [review]
lgtm

**Review by appsdesh — CHANGES_REQUESTED** [review]
This does not address the case where Transmitter addresses multiple versions of the spec

**Comment by tulshi** [comment]
> This does not address the case where Transmitter supports multiple versions of the spec

As we discussed a service offering multiple versions will use different metadata for each version it supports

**Comment by appsdesh** [comment]
> > This does not address the case where Transmitter supports multiple versions of the spec
> 
> As we discussed a service offering multiple versions will use different metadata for each version it supports

We have a single metadata URL `/.well-known/ssf-configurations` defined in the spec. In case of multiple version support, the transmitter should be able to publish version specific URLs.

**Comment by tulshi** [comment]
> > > This does not address the case where Transmitter supports multiple versions of the spec
> > 
> > 
> > As we discussed a service offering multiple versions will use different metadata for each version it supports
> 
> We have a single metadata URL `/.well-known/ssf-configurations` defined in the spec. In case of multiple version support, the transmitter should be able to publish version specific URLs.

Can the transmitter base URL be different for Transmitters supporting different versions? e.g. "transmitter1.myco.com/.well-known/.../" and "transmitter2.myco.com/.well-known/..."?

**Comment by appsdesh** [comment]
> > > > This does not address the case where Transmitter supports multiple versions of the spec
> > > 
> > > 
> > > As we discussed a service offering multiple versions will use different metadata for each version it supports
> > 
> > 
> > We have a single metadata URL `/.well-known/ssf-configurations` defined in the spec. In case of multiple version support, the transmitter should be able to publish version specific URLs.
> 
> Can the transmitter base URL be different for Transmitters supporting different versions? e.g. "transmitter1.myco.com/.well-known/.../" and "transmitter2.myco.com/.well-known/..."?

This is an SSF versioning concern, not really related to the base URL. Hence the solution should not be base URL dependent.

I propose following option -
1. Bake in the version in the well-known URL
2. `/.well-known/ssf-configurations` will always show the latest version then transmitter supports. Receivers can fetch earlier versions by query params etc.

**Comment by tulshi** [comment]
> > > > > This does not address the case where Transmitter supports multiple versions of the spec
> > > > 
> > > > 
> > > > As we discussed a service offering multiple versions will use different metadata for each version it supports
> > > 
> > > 
> > > We have a single metadata URL `/.well-known/ssf-configurations` defined in the spec. In case of multiple version support, the transmitter should be able to publish version specific URLs.
> > 
> > 
> > Can the transmitter base URL be different for Transmitters supporting different versions? e.g. "transmitter1.myco.com/.well-known/.../" and "transmitter2.myco.com/.well-known/..."?
> 
> This is an SSF versioning concern, not really related to the base URL. Hence the solution should not be base URL dependent.
> 
> I propose following option -
> 
> 1. Bake in the version in the well-known URL
> 2. `/.well-known/ssf-configurations` will always show the latest version then transmitter supports. Receivers can fetch earlier versions by query params etc.

I'm not sure how this will help Receivers that need to work with a specific version. Is the intent here that Receivers should be able to discover a particular version supported by the Transmitter? If so, it will require a lot more structure around how specific versions can be discovered. The approach the current PR takes is the discovery of a specific version the Transmitter supports is outside the scope of this draft. My suggestion to use different domain names / base URLs, is similar to the option 1 you are suggesting.

**Comment by appsdesh** [comment]
> > > > > > This does not address the case where Transmitter supports multiple versions of the spec
> > > > > 
> > > > > 
> > > > > As we discussed a service offering multiple versions will use different metadata for each version it supports
> > > > 
> > > > 
> > > > We have a single metadata URL `/.well-known/ssf-configurations` defined in the spec. In case of multiple version support, the transmitter should be able to publish version specific URLs.
> > > 
> > > 
> > > Can the transmitter base URL be different for Transmitters supporting different versions? e.g. "transmitter1.myco.com/.well-known/.../" and "transmitter2.myco.com/.well-known/..."?
> > 
> > 
> > This is an SSF versioning concern, not really related to the base URL. Hence the solution should not be base URL dependent.
> > I propose following option -
> > 
> > 1. Bake in the version in the well-known URL
> > 2. `/.well-known/ssf-configurations` will always show the latest version then transmitter supports. Receivers can fetch earlier versions by query params etc.
> 
> I'm not sure how this will help Receivers that need to work with a specific version. Is the intent here that Receivers should be able to discover a particular version supported by the Transmitter? If so, it will require a lot more structure around how specific versions can be discovered. The approach the current PR takes is the discovery of a specific version the Transmitter supports is outside the scope of this draft. My suggestion to use different domain names / base URLs, is similar to the option 1 you are suggesting.

The main concern is where would transmitter publish version based transmitter metadata in the absence of ability to change base URL. The metadata doc is version specific, ie. every new version of the spec may have different metadata JSON. This version is not really a property of "the transmitter" as transmitters can support multiple versions.

In that case, as suggested in my option #1, should we have metadata URL as `/.well-known/v1.0/ssf-configurations`

**Comment by tulshi** [comment]
> > > > > > > This does not address the case where Transmitter supports multiple versions of the spec
> > > > > > 
> > > > > > 
> > > > > > As we discussed a service offering multiple versions will use different metadata for each version it supports
> > > > > 
> > > > > 
> > > > > We have a single metadata URL `/.well-known/ssf-configurations` defined in the spec. In case of multiple version support, the transmitter should be able to publish version specific URLs.
> > > > 
> > > > 
> > > > Can the transmitter base URL be different for Transmitters supporting different versions? e.g. "transmitter1.myco.com/.well-known/.../" and "transmitter2.myco.com/.well-known/..."?
> > > 
> > > 
> > > This is an SSF versioning concern, not really related to the base URL. Hence the solution should not be base URL dependent.
> > > I propose following option -
> > > 
> > > 1. Bake in the version in the well-known URL
> > > 2. `/.well-known/ssf-configurations` will always show the latest version then transmitter supports. Receivers can fetch earlier versions by query params etc.
> > 
> > 
> > I'm not sure how this will help Receivers that need to work with a specific version. Is the intent here that Receivers should be able to discover a particular version supported by the Transmitter? If so, it will require a lot more structure around how specific versions can be discovered. The approach the current PR takes is the discovery of a specific version the Transmitter supports is outside the scope of this draft. My suggestion to use different domain names / base URLs, is similar to the option 1 you are suggesting.
> 
> The main concern is where would transmitter publish version based transmitter metadata in the absence of ability to change base URL. The metadata doc is version specific, ie. every new version of the spec may have different metadata JSON. This version is not really a property of "the transmitter" as transmitters can support multiple versions.
> 
> In that case, as suggested in my option #1, should we have metadata URL as `/.well-known/v1.0/ssf-configurations`

Based on the content of section "6.2", the "/.well-known/ssf-configuration" needs to be inserted between the host component and the path component of the issuer. So if you have a Transmitter that supports, say v2.0 of the spec, then you can define the issuer to be "https://transmitter.myco.com/v2.0". The well-known URL will be: "https://transmitter.myco.com/.well-known/ssf-configuration/v2.0". Does that work? This does not require us to specify anything in the draft.

**Comment by appsdesh** [comment]
> > > > > > > > This does not address the case where Transmitter supports multiple versions of the spec
> > > > > > > 
> > > > > > > 
> > > > > > > As we discussed a service offering multiple versions will use different metadata for each version it supports
> > > > > > 
> > > > > > 
> > > > > > We have a single metadata URL `/.well-known/ssf-configurations` defined in the spec. In case of multiple version support, the transmitter should be able to publish version specific URLs.
> > > > > 
> > > > > 
> > > > > Can the transmitter base URL be different for Transmitters supporting different versions? e.g. "transmitter1.myco.com/.well-known/.../" and "transmitter2.myco.com/.well-known/..."?
> > > > 
> > > > 
> > > > This is an SSF versioning concern, not really related to the base URL. Hence the solution should not be base URL dependent.
> > > > I propose following option -
> > > > 
> > > > 1. Bake in the version in the well-known URL
> > > > 2. `/.well-known/ssf-configurations` will always show the latest version then transmitter supports. Receivers can fetch earlier versions by query params etc.
> > > 
> > > 
> > > I'm not sure how this will help Receivers that need to work with a specific version. Is the intent here that Receivers should be able to discover a particular version supported by the Transmitter? If so, it will require a lot more structure around how specific versions can be discovered. The approach the current PR takes is the discovery of a specific version the Transmitter supports is outside the scope of this draft. My suggestion to use different domain names / base URLs, is similar to the option 1 you are suggesting.
> > 
> > 
> > The main concern is where would transmitter publish version based transmitter metadata in the absence of ability to change base URL. The metadata doc is version specific, ie. every new version of the spec may have different metadata JSON. This version is not really a property of "the transmitter" as transmitters can support multiple versions.
> > In that case, as suggested in my option #1, should we have metadata URL as `/.well-known/v1.0/ssf-configurations`
> 
> Based on the content of section "6.2", the "/.well-known/ssf-configuration" needs to be inserted between the host component and the path component of the issuer. So if you have a Transmitter that supports, say v2.0 of the spec, then you can define the issuer to be "https://transmitter.myco.com/v2.0". The well-known URL will be: "https://transmitter.myco.com/.well-known/ssf-configuration/v2.0". Does that work? This does not require us to specify anything in the draft.

1. Your proposal "https://transmitter.myco.com/v2.0" will be at odds with [RFC8615](https://datatracker.ietf.org/doc/html/rfc8615#section-3). 
2. I feel, another option like `https://transmitter.myco.com/.well-known/ssf-configuration/v2.0` should be listed in the spec, similar to [6.2.1](https://openid.github.io/sharedsignals/openid-sharedsignals-framework-1_0.html#section-6.2.1) that talks about `issuer1` etc. else `issuer` definition and `version` definition in the URL will collide.

**Comment by tulshi** [comment]
> > > > > > > > > This does not address the case where Transmitter supports multiple versions of the spec
> > > > > > > > 
> > > > > > > > 
> > > > > > > > As we discussed a service offering multiple versions will use different metadata for each version it supports
> > > > > > > 
> > > > > > > 
> > > > > > > We have a single metadata URL `/.well-known/ssf-configurations` defined in the spec. In case of multiple version support, the transmitter should be able to publish version specific URLs.
> > > > > > 
> > > > > > 
> > > > > > Can the transmitter base URL be different for Transmitters supporting different versions? e.g. "transmitter1.myco.com/.well-known/.../" and "transmitter2.myco.com/.well-known/..."?
> > > > > 
> > > > > 
> > > > > This is an SSF versioning concern, not really related to the base URL. Hence the solution should not be base URL dependent.
> > > > > I propose following option -
> > > > > 
> > > > > 1. Bake in the version in the well-known URL
> > > > > 2. `/.well-known/ssf-configurations` will always show the latest version then transmitter supports. Receivers can fetch earlier versions by query params etc.
> > > > 
> > > > 
> > > > I'm not sure how this will help Receivers that need to work with a specific version. Is the intent here that Receivers should be able to discover a particular version supported by the Transmitter? If so, it will require a lot more structure around how specific versions can be discovered. The approach the current PR takes is the discovery of a specific version the Transmitter supports is outside the scope of this draft. My suggestion to use different domain names / base URLs, is similar to the option 1 you are suggesting.
> > > 
> > > 
> > > The main concern is where would transmitter publish version based transmitter metadata in the absence of ability to change base URL. The metadata doc is version specific, ie. every new version of the spec may have different metadata JSON. This version is not really a property of "the transmitter" as transmitters can support multiple versions.
> > > In that case, as suggested in my option #1, should we have metadata URL as `/.well-known/v1.0/ssf-configurations`
> > 
> > 
> > Based on the content of section "6.2", the "/.well-known/ssf-configuration" needs to be inserted between the host component and the path component of the issuer. So if you have a Transmitter that supports, say v2.0 of the spec, then you can define the issuer to be "https://transmitter.myco.com/v2.0". The well-known URL will be: "https://transmitter.myco.com/.well-known/ssf-configuration/v2.0". Does that work? This does not require us to specify anything in the draft.
> 
> 1. Your proposal "https://transmitter.myco.com/v2.0" will be at odds with [RFC8615](https://datatracker.ietf.org/doc/html/rfc8615#section-3).
> 2. I feel, another option like `https://transmitter.myco.com/.well-known/ssf-configuration/v2.0` should be listed in the spec, similar to [6.2.1](https://openid.github.io/sharedsignals/openid-sharedsignals-framework-1_0.html#section-6.2.1) that talks about `issuer1` etc. else `issuer` definition and `version` definition in the URL will collide.

Note that the "https://transmitter.myco.com/v2.0" is the issuer value, and not the well known URL. The well-known URL will be https://transmitter.myco.com/.well-known/ssf-configuration/v2.0 based on how section 6.2 of our SSF draft specifies the well known URL to be constructed. Note that "v2.0" is just an example. A Transmitter can choose the `iss` value: "https://transmitter.myco.com/best-version", and the well-known URL in that case will be "https://transmitter.myco.com/.well-known/ssf-configuration/best-version".

If I read the part of RFC8615, I am unable to understand how this scheme conflicts with that RFC. Please clarify.

**Review by appsdesh — APPROVED** [review]

**Comment by appsdesh** [comment]
> > > > > > > > > > This does not address the case where Transmitter supports multiple versions of the spec
> > > > > > > > > 
> > > > > > > > > 
> > > > > > > > > As we discussed a service offering multiple versions will use different metadata for each version it supports
> > > > > > > > 
> > > > > > > > 
> > > > > > > > We have a single metadata URL `/.well-known/ssf-configurations` defined in the spec. In case of multiple version support, the transmitter should be able to publish version specific URLs.
> > > > > > > 
> > > > > > > 
> > > > > > > Can the transmitter base URL be different for Transmitters supporting different versions? e.g. "transmitter1.myco.com/.well-known/.../" and "transmitter2.myco.com/.well-known/..."?
> > > > > > 
> > > > > > 
> > > > > > This is an SSF versioning concern, not really related to the base URL. Hence the solution should not be base URL dependent.
> > > > > > I propose following option -
> > > > > > 
> > > > > > 1. Bake in the version in the well-known URL
> > > > > > 2. `/.well-known/ssf-configurations` will always show the latest version then transmitter supports. Receivers can fetch earlier versions by query params etc.
> > > > > 
> > > > > 
> > > > > I'm not sure how this will help Receivers that need to work with a specific version. Is the intent here that Receivers should be able to discover a particular version supported by the Transmitter? If so, it will require a lot more structure around how specific versions can be discovered. The approach the current PR takes is the discovery of a specific version the Transmitter supports is outside the scope of this draft. My suggestion to use different domain names / base URLs, is similar to the option 1 you are suggesting.
> > > > 
> > > > 
> > > > The main concern is where would transmitter publish version based transmitter metadata in the absence of ability to change base URL. The metadata doc is version specific, ie. every new version of the spec may have different metadata JSON. This version is not really a property of "the transmitter" as transmitters can support multiple versions.
> > > > In that case, as suggested in my option #1, should we have metadata URL as `/.well-known/v1.0/ssf-configurations`
> > > 
> > > 
> > > Based on the content of section "6.2", the "/.well-known/ssf-configuration" needs to be inserted between the host component and the path component of the issuer. So if you have a Transmitter that supports, say v2.0 of the spec, then you can define the issuer to be "https://transmitter.myco.com/v2.0". The well-known URL will be: "https://transmitter.myco.com/.well-known/ssf-configuration/v2.0". Does that work? This does not require us to specify anything in the draft.
> > 
> > 
> > 
> > 1. Your proposal "https://transmitter.myco.com/v2.0" will be at odds with [RFC8615](https://datatracker.ietf.org/doc/html/rfc8615#section-3).
> > 2. I feel, another option like `https://transmitter.myco.com/.well-known/ssf-configuration/v2.0` should be listed in the spec, similar to [6.2.1](https://openid.github.io/sharedsignals/openid-sharedsignals-framework-1_0.html#section-6.2.1) that talks about `issuer1` etc. else `issuer` definition and `version` definition in the URL will collide.
> 
> Note that the "https://transmitter.myco.com/v2.0" is the issuer value, and not the well known URL. The well-known URL will be https://transmitter.myco.com/.well-known/ssf-configuration/v2.0 based on how section 6.2 of our SSF draft specifies the well known URL to be constructed. Note that "v2.0" is just an example. A Transmitter can choose the `iss` value: "https://transmitter.myco.com/best-version", and the well-known URL in that case will be "https://transmitter.myco.com/.well-known/ssf-configuration/best-version".
> 
> If I read the part of RFC8615, I am unable to understand how this scheme conflicts with that RFC. Please clarify.

Thanks, the example helps!

**Pull request merged**